### ISSUE TITLE
Trouble adding data sources

### ISSUE THREAD

**Issue by npateman** [getredash/redash#832]
Hi there,

When I add a datasource, it says it's saved it, with a green tick, but if I open it back up again to modify it, all of the settings have gone.

Nick.

**Comment by arikfr** [comment]
What data source type is it? Can you add a screenshot?

**Comment by npateman** [comment]
It's a mysql one...

![image](https://cloud.githubusercontent.com/assets/1122645/13073161/0ad107da-d495-11e5-84cb-1495cef63ab1.png)

That's what it looks like when I go back into it after saving.

**Comment by arikfr** [comment]
If you enter it again and save, you get the same result?

What version of Re:dash are you running? Were there any local modifications to it?

**Comment by npateman** [comment]
I just added property by property, one at a time and it kept it.  Then tested it but it failed so i went back in, disabled ssl, saved it, now it's lost everything again.

No mods, installed from an AMI instance, version 0.9.1+b1377

**Comment by arikfr** [comment]
What do you get when you hit the API directly?
If you data source id is 1 (you can get the id from the URL), then the API URL will be: `/api/data_sources/1`.

**Comment by npateman** [comment]
Just trying out a different datasource type, apparently I was using the wrong one, will let you know if that fixes it.  Sorry for being vague!

**Comment by npateman** [comment]
Now this is what happened to me last time, all data sources have vanished, so I am using the manage.py script to delete the one I created so that I can start again but I'm getting this error from the script.

```
ubuntu@ip-172-31-44-151:/opt/redash/current$ sudo -u redash bin/run ./manage.py ds delete mydatabase
[2016-02-16 10:54:14,156][PID:29588][WARNING][redash.query_runner] MQL query runner enabled but not supported, not registering. Either disable or install missing dependencies.
[2016-02-16 10:54:14,200][PID:29588][WARNING][redash.query_runner] Oracle query runner enabled but not supported, not registering. Either disable or install missing dependencies.
Deleting data source: mydatabase (id=10)
[2016-02-16 10:54:14,416][PID:29588][ERROR][peewee] DELETE FROM "data_sources" WHERE ("id" = %s) [10]
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/playhouse/postgres_ext.py", line 360, in execute_sql
    cursor.execute(sql, params or ())
IntegrityError: update or delete on table "data_sources" violates foreign key constraint "data_source_groups_data_source_id_fkey" on table "data_source_groups"
DETAIL:  Key (id)=(10) is still referenced from table "data_source_groups".

Traceback (most recent call last):
  File "/opt/redash/redash.0.9.1.b1377/manage.py", line 58, in <module>
    manager.run()
  File "/usr/local/lib/python2.7/dist-packages/flask_script/__init__.py", line 405, in run
    result = self.handle(sys.argv[0], sys.argv[1:])
  File "/usr/local/lib/python2.7/dist-packages/flask_script/__init__.py", line 384, in handle
    return handle(app, *positional_args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/flask_script/commands.py", line 145, in handle
    return self.run(*args, **kwargs)
  File "/opt/redash/redash.0.9.1.b1377/redash/cli/data_sources.py", line 98, in delete
    data_source.delete_instance()
  File "/usr/local/lib/python2.7/dist-packages/peewee.py", line 4195, in delete_instance
    return self.delete().where(self._pk_expr()).execute()
  File "/usr/local/lib/python2.7/dist-packages/peewee.py", line 2884, in execute
    return self.database.rows_affected(self._execute())
  File "/opt/redash/redash.0.9.1.b1377/redash/metrics/database.py", line 50, in metered_execute
    result = real_execute(self, *args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/peewee.py", line 2371, in _execute
    return self.database.execute_sql(sql, params, self.require_commit)
  File "/opt/redash/redash.0.9.1.b1377/redash/metrics/database.py", line 22, in execute_sql
    result = super(MeteredPostgresqlExtDatabase, self).execute_sql(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/playhouse/postgres_ext.py", line 369, in execute_sql
    self.commit()
  File "/usr/local/lib/python2.7/dist-packages/peewee.py", line 2922, in __exit__
    reraise(new_type, new_type(*exc_value.args), traceback)
  File "/usr/local/lib/python2.7/dist-packages/playhouse/postgres_ext.py", line 360, in execute_sql
    cursor.execute(sql, params or ())
peewee.IntegrityError: update or delete on table "data_sources" violates foreign key constraint "data_source_groups_data_source_id_fkey" on table "data_source_groups"
DETAIL:  Key (id)=(10) is still referenced from table "data_source_groups".
```

**Comment by arikfr** [comment]
About the CLI issue, see my comment here about how to delete a data source from the CLI: https://github.com/getredash/redash/issues/814#issuecomment-181503175

Note that it was fixed in #815 and it's part of the next release.

**Comment by npateman** [comment]
Great, that worked, deleting the mysql datasource using the cli and entering a new one in the gui fixed the issue.  Many thanks.
